src/include/parser/postgresparser.h

```diff
@@ -36,7 +53,7 @@ class PostgresParser {
    * @param query_string query string to be parsed
    * @return unique pointer to parse tree
    */
-  std::vector<std::unique_ptr<parser::SQLStatement>> BuildParseTree(const std::string &query_string);
+  ParseResult BuildParseTree(const std::string &query_string);
```

Comment:
Good call in abstracting this away into its own object, makes it much easier to understand

Reply:
I'm a little unhappy about how it gets passed around through every single function call..